packages/workspace/src/core/target-project-locator.ts

```diff
@@ -27,28 +41,45 @@ export class TargetProjectLocator {
     importExpr: string,
     filePath: string,
     npmScope: string
-  ) {
+  ): string {
     const normalizedImportExpr = importExpr.split('#')[0];
 
-    const resolvedModule = resolveModuleByImport(
-      normalizedImportExpr,
-      filePath
-    );
-
-    return this._sortedNodeNames.find((projectName) => {
-      const p = this.nodes[projectName];
+    if (isRelativePath(normalizedImportExpr)) {
+      const resolvedModule = join(dirname(filePath), normalizedImportExpr);
+      return this.findProjectOfResolvedModule(resolvedModule);
+    }
 
-      if (!isWorkspaceProject(p)) {
-        return false;
+    if (this.paths && this.paths[normalizedImportExpr]) {
+      for (let p of this.paths[normalizedImportExpr]) {
+        const maybeResolvedProject = this.findProjectOfResolvedModule(p);
+        if (maybeResolvedProject) {
+          return maybeResolvedProject;
+        }
       }
+    }
 
-      if (resolvedModule && resolvedModule.startsWith(p.data.root)) {
-        return true;
-      } else {
-        const normalizedRoot = normalizedProjectRoot(p);
-        const projectImport = `@${npmScope}/${normalizedRoot}`;
-        return normalizedImportExpr.startsWith(projectImport);
-      }
+    const resolvedModule = this.cache.has(normalizedImportExpr)
+      ? this.cache.get(normalizedImportExpr)
+      : resolveModuleByImport(normalizedImportExpr, filePath);
+
+    this.cache.set(normalizedImportExpr, resolvedModule);
+    if (resolvedModule) {
+      return this.findProjectOfResolvedModule(resolvedModule);
+    }
+
+    const importedProject = this.sortedWorkspaceProjects.find((p) => {
+      const projectImport = `@${npmScope}/${p.data.normalizedRoot}`;
+      return normalizedImportExpr.startsWith(projectImport);
     });
+
+    return importedProject?.name;
+  }
+
+  private findProjectOfResolvedModule(resolvedModule: string) {
+    const importedProject = this.sortedWorkspaceProjects.find((p) => {
+      return resolvedModule.startsWith(p.data.root);
+    });
+
+    return importedProject?.name;
```

Comment:
This sort of implies importedProject can be null :) Maybe if it's "null", we can fall back to using typescript?

Reply:
Typescript was already used to resolve the module of the import. This is null when the file get's imported from somewhere which is not a workspace project.